test-harness/src/test/groovy/com/netflix/conductor/test/integration/JsonJQTransformSpec.groovy

```diff
@@ -84,4 +85,66 @@ class JsonJQTransformSpec extends AbstractSpecification {
             tasks[0].reasonForIncompletion as String == "Cannot index string with string \"array\""
         }
     }
+
+    /**
+     * Given the following invalid input JSON
+     *{*   "in1": "a",
+     *   "in2": "b"
+     *}* using the same query from the success test, jq will try to get in1['array']
+     * and fail since 'in1' is a string.
+     *
+     * Re-run failed system task with the following valid input JSON will fix the workflow
+     *{*   "in1": {*     "array": [ "a", "b" ]
+     *},
+     *   "in2": {*     "array": [ "c", "d" ]
+     *}*}* expect the workflow task to transform to following result:
+     *{*     out: [ "a", "b", "c", "d" ]
+     *}
+     */
+    def "Test rerun workflow with failed json jq transform task"() {
+        given: "workflow input"
+        def invalidInput = new HashMap()
+        invalidInput['in1'] = "a"
+        invalidInput['in2'] = "b"
+
+        def validInput = new HashMap()
+        def input = new HashMap()
+        input['in1'] = new HashMap()
+        input['in1']['array'] = ["a", "b"]
+        input['in2'] = new HashMap()
+        input['in2']['array'] = ["c", "d"]
+        validInput['input'] = input
+        validInput['queryExpression'] = '.input as $_ | { out: ($_.in1.array + $_.in2.array) }'
+
+        when: "workflow which has the json jq transform task started"
+        def workflowInstanceId = workflowExecutor.startWorkflow(JSON_JQ_TRANSFORM_WF, 1,
+                '', invalidInput, null, null, null)
+
+        then: "verify that the workflow and task failed with expected error"
+        with(workflowExecutionService.getExecutionStatus(workflowInstanceId, true)) {
+            status == Workflow.WorkflowStatus.FAILED
+            tasks.size() == 1
+            tasks[0].status == Task.Status.FAILED
+            tasks[0].taskType == 'JSON_JQ_TRANSFORM'
+            tasks[0].reasonForIncompletion as String == "Cannot index string with string \"array\""
+        }
+
+        when: "workflow which has the json jq transform task reran"
+        def reRunWorkflowRequest = new RerunWorkflowRequest()
+        reRunWorkflowRequest.reRunFromWorkflowId = workflowInstanceId
+        def reRunTaskId = workflowExecutionService.getExecutionStatus(workflowInstanceId, true).tasks[0].taskId
+        reRunWorkflowRequest.reRunFromTaskId = reRunTaskId
+        reRunWorkflowRequest.taskInput = validInput
+
+        workflowExecutor.rerun(reRunWorkflowRequest)
+
+        then: "verify that the workflow and task are completed with expected output"
+        with(workflowExecutionService.getExecutionStatus(workflowInstanceId, true)) {
+            status == Workflow.WorkflowStatus.COMPLETED
+            tasks.size() == 1
+            tasks[0].status == Task.Status.COMPLETED
+            tasks[0].taskType == 'JSON_JQ_TRANSFORM'
+            tasks[0].outputData as String == "[result:[out:[a, b, c, d]], resultList:[[out:[a, b, c, d]]]]"
```

Comment:
same as above

Reply:
updated as well, the `as String` is necessary for outputData though as it's a map, so kept it, but used single quotes for the match string instead

Reply:
I missed that it's a map here. In that case, it makes sense to assert the keys of the map rather than converting it to s String and asserting that. The assertion will fail, if the `toString()` logic in Map changes, its unlikely, but possible.

Reply:
I think we can rely on Java's implementation of `toSTring()` of the Map. It's kind of cumbersome to write a loop through the map to assert each key/value pair. What do you think?

Reply:
Verifying the contents of a Map using its `toString()` is not ideal. It's also harder to understand and refactor. The goal of the test is not to verify if `jackson-jq` is doing its work, so you could 

1. write a simpler `jq` expression that is easy to verify or 
2. remove the `outputData` assertion altogether or
3. assert that `output.result != null` and `output.resultList != null`

Reply:
Make sense, didn't realize the outputData result is JsonNode. yea, the purpose of the test isn't for those kind of checks. Refactored to only do basic assertions.